pennylane/capture/base_interpreter.py

```diff
@@ -311,20 +314,21 @@ def eval(self, jaxpr: "jax.core.Jaxpr", consts: Sequence, *args) -> list:
             self._env[constvar] = const
 
         for eqn in jaxpr.eqns:
+            primitive = eqn.primitive
+            custom_handler = self._primitive_registrations.get(primitive, None)
 
-            custom_handler = self._primitive_registrations.get(eqn.primitive, None)
             if custom_handler:
                 invals = [self.read(invar) for invar in eqn.invars]
                 outvals = custom_handler(self, *invals, **eqn.params)
-            elif isinstance(eqn.outvars[0].aval, AbstractOperator):
+            elif isinstance(primitive, QmlPrimitive) and primitive.p_type.value == "operator":
```

Comment:
```suggestion
            elif getattr(primitive, "p_type", "") == "operator":
```
or something like that?

Reply:
Updated